File: memory_trainer.rb
# memory_trainer.rb — Ruby версия

COLORS = {
  'red' => "\e[31mкрасный\e[0m",
  'blue' => "\e[34mсиний\e[0m",
  'green' => "\e[32mзелёный\e[0m",
  'yellow' => "\e[33mжёлтый\e[0m",
  'magenta' => "\e[35mфиолетовый\e[0m",
  'cyan' => "\e[36mголубой\e[0m",
  'white' => "\e[37mбелый\e[0m"
}
COLOR_NAMES = COLORS.keys

def clear_screen
  system('clear') || system('cls')
end

def play
  score = 0
  level = 1
  sequence = []

  loop do
    new_color = COLOR_NAMES.sample
    sequence << new_color

    clear_screen
    puts "🎨 Memory Trainer (Colors) (Ruby)"
    puts "Уровень #{level}: Запомните #{sequence.size} цветов!"
    print "Показываю: "
    sequence.each { |c| print COLORS[c] + " " }
    puts "\n" + "=" * 30
    sleep 1.5
    clear_screen

    puts "Введите #{sequence.size} цветов через пробел:"
    print "> "
    input = gets.chomp.strip.downcase
    user_colors = input.split

    if user_colors.size != sequence.size
      puts "\e[31m❌ Нужно ввести #{sequence.size} цветов. Попробуйте ещё раз.\e[0m"
      sleep 1
      next
    end

    correct = user_colors == sequence
    if correct
      puts "\e[32m✅ Правильно! +1 очко.\e[0m"
      score += 1
      level += 1
    else
      puts "\e[31m❌ Неправильно. Правильная последовательность: #{sequence.join(' ')}\e[0m"
      puts "Ваш счёт: #{score}"
      break
    end
    sleep 1
  end
  puts "\nИгра окончена! Счёт: #{score}"
end

play

File: memory_trainer.py


### 1. `memory_trainer.py` (Python)

```python
# memory_trainer.py — Python версия

import random
import sys
import time
import os
from colorama import init, Fore, Style

init(autoreset=True)

COLORS = {
    'red': Fore.RED + 'красный' + Style.RESET_ALL,
    'blue': Fore.BLUE + 'синий' + Style.RESET_ALL,
    'green': Fore.GREEN + 'зелёный' + Style.RESET_ALL,
    'yellow': Fore.YELLOW + 'жёлтый' + Style.RESET_ALL,
    'magenta': Fore.MAGENTA + 'фиолетовый' + Style.RESET_ALL,
    'cyan': Fore.CYAN + 'голубой' + Style.RESET_ALL,
    'white': Fore.WHITE + 'белый' + Style.RESET_ALL
}
COLOR_NAMES = list(COLORS.keys())

def clear_screen():
    os.system('cls' if os.name == 'nt' else 'clear')

def get_user_input(prompt):
    return input(prompt).strip().lower().split()

def play_game():
    score = 0
    level = 1
    sequence = []

    while True:
        # Генерируем новый цвет для последовательности
        seq_len = level + 2  # начинаем с 3 цветов
        new_color = random.choice(COLOR_NAMES)
        sequence.append(new_color)

        # Показываем последовательность
        clear_screen()
        print(f"🎨 Memory Trainer (Colors) (Python)")
        print(f"Уровень {level}: Запомните {len(sequence)} цветов!")
        print("Показываю: ", end="")
        for c in sequence:
            print(COLORS[c], end=" ")
        print("\n" + "=" * 30)
        time.sleep(1.5)
        clear_screen()

        # Запрашиваем ввод
        print(f"Введите {len(sequence)} цветов через пробел:")
        user_input = get_user_input("> ")

        # Проверяем
        if len(user_input) != len(sequence):
            print(Fore.RED + f"❌ Нужно ввести {len(sequence)} цветов. Попробуйте ещё раз.")
            time.sleep(1)
            continue

        # Сравниваем (игнорируем регистр)
        correct = True
        for i, expected in enumerate(sequence):
            if user_input[i] != expected:
                correct = False
                break

        if correct:
            print(Fore.GREEN + f"✅ Правильно! +1 очко.")
            score += 1
            level += 1
        else:
            print(Fore.RED + f"❌ Неправильно. Правильная последовательность: " + " ".join(sequence))
            print(f"Ваш счёт: {score}")
            break

        # Небольшая пауза
        time.sleep(1)

    print(f"\nИгра окончена! Счёт: {score}")

def main():
    try:
        play_game()
    except KeyboardInterrupt:
        print("\nВыход...")

if __name__ == "__main__":
    main()
